sycl/source/handler.cpp

```diff
@@ -769,15 +769,19 @@ event handler::finalize() {
           nullptr, impl->MExecGraph, std::move(impl->CGData)));
 
     } else {
-      event GraphCompletionEvent =
-          impl->MExecGraph->enqueue(impl->get_queue(), std::move(impl->CGData));
-
+      detail::queue_impl &Queue = impl->get_queue();
+      bool DiscardEvent = !impl->MEventNeeded &&
+                          Queue.supportsDiscardingPiEvents() &&
+                          !impl->MExecGraph->containsHostTask();
+      detail::EventImplPtr GraphCompletionEvent = impl->MExecGraph->enqueue(
+          Queue, std::move(impl->CGData), !DiscardEvent);
 #ifdef __INTEL_PREVIEW_BREAKING_CHANGES
-      MLastEvent = getSyclObjImpl(GraphCompletionEvent);
+      return GraphCompletionEvent;
```

Comment:
Why aren't we setting `MLastEvent` anymore? \+ @slawekptak

Reply:
Thanks for the input @slawekptak. If that's the case, I would prefer to keep the code as is. I don't expect the graph enqueue path to ever make use of `MLastEvent` member variable. Does that sounds reasonable @aelovikov-intel ?